{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8798fcc7-78b8-4e2d-be28-4f504d2105fb",
   "metadata": {},
   "source": [
    "# Create Human Trisome Project PLIER model"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd280251-a5e1-4ed1-a255-3ce87ab5faa4",
   "metadata": {},
   "source": [
    "# Load libraries/modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a695300b-c727-4fa6-a4fd-64d297fab268",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "here() starts at /home/msubirana/Documents/pivlab/plier_recount3\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ─────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.5\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.1     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.3     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.2     \n",
      "── \u001b[1mConflicts\u001b[22m ───────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32m%within%()\u001b[39m masks \u001b[34mIRanges\u001b[39m::%within%()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcollapse()\u001b[39m     masks \u001b[34mIRanges\u001b[39m::collapse()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m      masks \u001b[34mBiobase\u001b[39m::combine(), \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcount()\u001b[39m        masks \u001b[34mmatrixStats\u001b[39m::count()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mdesc()\u001b[39m         masks \u001b[34mIRanges\u001b[39m::desc()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m       masks \u001b[34mS4Vectors\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m       masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m        masks \u001b[34mS4Vectors\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m          masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m   masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mreduce()\u001b[39m       masks \u001b[34mGenomicRanges\u001b[39m::reduce(), \u001b[34mIRanges\u001b[39m::reduce()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mrename()\u001b[39m       masks \u001b[34mS4Vectors\u001b[39m::rename()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond()\u001b[39m   masks \u001b[34mS4Vectors\u001b[39m::second()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond<-()\u001b[39m masks \u001b[34mS4Vectors\u001b[39m::second<-()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mselect()\u001b[39m       masks \u001b[34mbiomaRt\u001b[39m::select()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mslice()\u001b[39m        masks \u001b[34mIRanges\u001b[39m::slice()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "\n",
      "Attaching package: ‘cowplot’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ggpubr’:\n",
      "\n",
      "    get_legend\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:lubridate’:\n",
      "\n",
      "    stamp\n",
      "\n",
      "\n",
      "Loading required package: RColorBrewer\n",
      "\n",
      "Loading required package: gplots\n",
      "\n",
      "\n",
      "Attaching package: ‘gplots’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:rtracklayer’:\n",
      "\n",
      "    space\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    space\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    space\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    lowess\n",
      "\n",
      "\n",
      "Loading required package: pheatmap\n",
      "\n",
      "Loading required package: glmnet\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:tidyr’:\n",
      "\n",
      "    expand, pack, unpack\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loaded glmnet 4.1-8\n",
      "\n",
      "Loading required package: knitr\n",
      "\n",
      "Loading required package: rsvd\n",
      "\n",
      "Loading required package: qvalue\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(biomaRt)\n",
    "library(here)\n",
    "library(biomaRt)\n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(rtracklayer)\n",
    "library(dplyr)\n",
    "library(GenomicRanges)\n",
    "library(ggpubr)\n",
    "library(cowplot)\n",
    "# load plier utils\n",
    "source(here::here('scripts/plier_util.R'))\n",
    "`%>%` <- dplyr::`%>%`\n",
    "library(PLIER)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "249ff9e0-ce35-4e9e-873a-34235c45aff7",
   "metadata": {},
   "source": [
    "# Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a5ab23a3-9aa6-43a9-b9d0-29f7d5d436fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "output_nb_path = here('output/nbs/create_human_trisome_project_plier_model/')\n",
    "dir.create(output_nb_path, showWarnings = FALSE)\n",
    "\n",
    "# Load PLIER pathway and cell type data\n",
    "data(bloodCellMarkersIRISDMAP)\n",
    "data(svmMarkers)\n",
    "data(canonicalPathways)\n",
    "data(chemgenPathways)\n",
    "data(oncogenicPathways)\n",
    "data(xCell)\n",
    "data(immunePathways)\n",
    "chr21_pathway <- readRDS(here::here('output/pathways/chr21_pathway.rds'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53ab5bb7-5f0b-4bc1-b074-f330fd01ddaf",
   "metadata": {},
   "source": [
    "## GSE190125 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a4a1fe97-264d-4472-8696-5770029c2642",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_matrix=here::here('data/GSE190125/GSE190125_Counts_for_GEO.txt')\n",
    "raw_gene_counts_GSE190125 <- read.table(counts_matrix, header = TRUE, sep = \"\\t\", check.names = FALSE)\n",
    "\n",
    "gene_counts_GSE190125 <- raw_gene_counts_GSE190125 %>%\n",
    " dplyr::select(-chr, -gene_type) %>%\n",
    "  pivot_wider(\n",
    "    names_from = SampleID,\n",
    "    values_from = raw_count,\n",
    "    values_fill = list(raw_count = 0)\n",
    "  ) %>%\n",
    "  dplyr::select(Geneid = EnsemblID, GeneSymbol = gene_name, everything())\n",
    "\n",
    "head(gene_counts_GSE190125)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58fbe04b-e3fa-46f6-be9e-9f5f233ad63a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter T21 samples\n",
    "path_metadata_GSE190125 <- here::here('data/GSE190125/metadata_GSE190125.csv')\n",
    "metadata_GSE190125 <- read.csv(path_metadata_GSE190125)\n",
    "colnames(metadata_GSE190125) <- c('Group', 'Sample')\t\n",
    "\n",
    "#t21_samples <- metadata_GSE190125 %>% \n",
    "#  filter(Group == \"T21\") %>%\n",
    "#  pull(Sample)\n",
    "\n",
    "gene_counts_T21 <- gene_counts_GSE190125 %>%\n",
    "  select(Geneid, GeneSymbol, matches(paste(t21_samples, collapse = \"|\")))\n",
    "\n",
    "head(gene_counts_T21)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "33847fd3-a058-431f-a057-6b42d363d500",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(GeneSymbol)`\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 400 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>HTP0001B2</th><th scope=col>HTP0005A3</th><th scope=col>HTP0012A2</th><th scope=col>HTP0015A4</th><th scope=col>HTP0017A4</th><th scope=col>HTP0018B3</th><th scope=col>HTP0019B2</th><th scope=col>HTP0022B2</th><th scope=col>HTP0023A2</th><th scope=col>HTP0025A3</th><th scope=col>⋯</th><th scope=col>HTP0664A</th><th scope=col>HTP0665A</th><th scope=col>HTP0666A</th><th scope=col>HTP0667A</th><th scope=col>HTP0668A</th><th scope=col>HTP0669B</th><th scope=col>HTP0672A</th><th scope=col>HTP0676A</th><th scope=col>HTP0706A</th><th scope=col>HTP0708A</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>A1BG</th><td> 1.213548628</td><td> 0.74541369</td><td> 0.599729304</td><td> 0.32731643</td><td> 1.3535745</td><td> 0.94879940</td><td> 0.6912469</td><td> 1.097659107</td><td> 0.6766134</td><td> 0.27325337</td><td>⋯</td><td> 0.255005402</td><td> 0.52592489</td><td> 0.28917116</td><td> 0.4338034</td><td> 1.0180578</td><td> 1.07863098</td><td> 0.66926351</td><td> 1.31704224</td><td> 0.422119153</td><td> 1.10667053</td></tr>\n",
       "\t<tr><th scope=row>A1BG.AS1</th><td>38.222498659</td><td>24.92606386</td><td>26.427416614</td><td>12.07818145</td><td>24.1682646</td><td>20.46200787</td><td>29.3621799</td><td>24.934104324</td><td>37.2498378</td><td>16.60207704</td><td>⋯</td><td>32.137704564</td><td>37.67020602</td><td>19.36197036</td><td>23.5854179</td><td>29.6142852</td><td>31.45654523</td><td>17.50423671</td><td>26.70369761</td><td>22.246704141</td><td>31.47242852</td></tr>\n",
       "\t<tr><th scope=row>A1CF</th><td> 0.003766156</td><td> 0.00000000</td><td> 0.003579259</td><td> 0.00000000</td><td> 0.0000000</td><td> 0.00000000</td><td> 0.0000000</td><td> 0.000000000</td><td> 0.0000000</td><td> 0.00000000</td><td>⋯</td><td> 0.000000000</td><td> 0.00000000</td><td> 0.00000000</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.00000000</td><td> 0.00000000</td><td> 0.00000000</td><td> 0.000000000</td><td> 0.00000000</td></tr>\n",
       "\t<tr><th scope=row>A2M</th><td> 0.040175335</td><td> 0.05952886</td><td> 0.025454407</td><td> 0.07095019</td><td> 0.1066928</td><td> 0.03079474</td><td> 0.1616818</td><td> 0.079310732</td><td> 0.1199986</td><td> 0.03769264</td><td>⋯</td><td> 0.052763272</td><td> 0.08370718</td><td> 0.06838003</td><td> 0.1077101</td><td> 0.0200616</td><td> 0.04734121</td><td> 0.09231839</td><td> 0.04844618</td><td> 0.005822723</td><td> 0.07964579</td></tr>\n",
       "\t<tr><th scope=row>A2M.AS1</th><td> 7.224822540</td><td> 7.02529435</td><td> 5.042429225</td><td>12.28791629</td><td> 5.7335597</td><td> 7.13868477</td><td> 9.2958055</td><td> 7.779616698</td><td>10.7898108</td><td> 6.35470746</td><td>⋯</td><td> 4.052397341</td><td> 2.16390527</td><td> 4.32314182</td><td> 9.3822274</td><td> 4.4814680</td><td> 6.95520729</td><td> 2.50756773</td><td> 9.90330650</td><td> 3.730342348</td><td> 8.72800822</td></tr>\n",
       "\t<tr><th scope=row>A2ML1</th><td> 0.000000000</td><td> 0.00000000</td><td> 0.004770096</td><td> 0.00000000</td><td> 0.0000000</td><td> 0.01731259</td><td> 0.0000000</td><td> 0.005404598</td><td> 0.0000000</td><td> 0.00000000</td><td>⋯</td><td> 0.008789078</td><td> 0.00000000</td><td> 0.00000000</td><td> 0.0000000</td><td> 0.0000000</td><td> 0.00000000</td><td> 0.00000000</td><td> 0.00000000</td><td> 0.004364659</td><td> 0.00000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 400 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & HTP0001B2 & HTP0005A3 & HTP0012A2 & HTP0015A4 & HTP0017A4 & HTP0018B3 & HTP0019B2 & HTP0022B2 & HTP0023A2 & HTP0025A3 & ⋯ & HTP0664A & HTP0665A & HTP0666A & HTP0667A & HTP0668A & HTP0669B & HTP0672A & HTP0676A & HTP0706A & HTP0708A\\\\\n",
       "\\hline\n",
       "\tA1BG &  1.213548628 &  0.74541369 &  0.599729304 &  0.32731643 &  1.3535745 &  0.94879940 &  0.6912469 &  1.097659107 &  0.6766134 &  0.27325337 & ⋯ &  0.255005402 &  0.52592489 &  0.28917116 &  0.4338034 &  1.0180578 &  1.07863098 &  0.66926351 &  1.31704224 &  0.422119153 &  1.10667053\\\\\n",
       "\tA1BG.AS1 & 38.222498659 & 24.92606386 & 26.427416614 & 12.07818145 & 24.1682646 & 20.46200787 & 29.3621799 & 24.934104324 & 37.2498378 & 16.60207704 & ⋯ & 32.137704564 & 37.67020602 & 19.36197036 & 23.5854179 & 29.6142852 & 31.45654523 & 17.50423671 & 26.70369761 & 22.246704141 & 31.47242852\\\\\n",
       "\tA1CF &  0.003766156 &  0.00000000 &  0.003579259 &  0.00000000 &  0.0000000 &  0.00000000 &  0.0000000 &  0.000000000 &  0.0000000 &  0.00000000 & ⋯ &  0.000000000 &  0.00000000 &  0.00000000 &  0.0000000 &  0.0000000 &  0.00000000 &  0.00000000 &  0.00000000 &  0.000000000 &  0.00000000\\\\\n",
       "\tA2M &  0.040175335 &  0.05952886 &  0.025454407 &  0.07095019 &  0.1066928 &  0.03079474 &  0.1616818 &  0.079310732 &  0.1199986 &  0.03769264 & ⋯ &  0.052763272 &  0.08370718 &  0.06838003 &  0.1077101 &  0.0200616 &  0.04734121 &  0.09231839 &  0.04844618 &  0.005822723 &  0.07964579\\\\\n",
       "\tA2M.AS1 &  7.224822540 &  7.02529435 &  5.042429225 & 12.28791629 &  5.7335597 &  7.13868477 &  9.2958055 &  7.779616698 & 10.7898108 &  6.35470746 & ⋯ &  4.052397341 &  2.16390527 &  4.32314182 &  9.3822274 &  4.4814680 &  6.95520729 &  2.50756773 &  9.90330650 &  3.730342348 &  8.72800822\\\\\n",
       "\tA2ML1 &  0.000000000 &  0.00000000 &  0.004770096 &  0.00000000 &  0.0000000 &  0.01731259 &  0.0000000 &  0.005404598 &  0.0000000 &  0.00000000 & ⋯ &  0.008789078 &  0.00000000 &  0.00000000 &  0.0000000 &  0.0000000 &  0.00000000 &  0.00000000 &  0.00000000 &  0.004364659 &  0.00000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 400 of type dbl\n",
       "\n",
       "| <!--/--> | HTP0001B2 | HTP0005A3 | HTP0012A2 | HTP0015A4 | HTP0017A4 | HTP0018B3 | HTP0019B2 | HTP0022B2 | HTP0023A2 | HTP0025A3 | ⋯ | HTP0664A | HTP0665A | HTP0666A | HTP0667A | HTP0668A | HTP0669B | HTP0672A | HTP0676A | HTP0706A | HTP0708A |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| A1BG |  1.213548628 |  0.74541369 |  0.599729304 |  0.32731643 |  1.3535745 |  0.94879940 |  0.6912469 |  1.097659107 |  0.6766134 |  0.27325337 | ⋯ |  0.255005402 |  0.52592489 |  0.28917116 |  0.4338034 |  1.0180578 |  1.07863098 |  0.66926351 |  1.31704224 |  0.422119153 |  1.10667053 |\n",
       "| A1BG.AS1 | 38.222498659 | 24.92606386 | 26.427416614 | 12.07818145 | 24.1682646 | 20.46200787 | 29.3621799 | 24.934104324 | 37.2498378 | 16.60207704 | ⋯ | 32.137704564 | 37.67020602 | 19.36197036 | 23.5854179 | 29.6142852 | 31.45654523 | 17.50423671 | 26.70369761 | 22.246704141 | 31.47242852 |\n",
       "| A1CF |  0.003766156 |  0.00000000 |  0.003579259 |  0.00000000 |  0.0000000 |  0.00000000 |  0.0000000 |  0.000000000 |  0.0000000 |  0.00000000 | ⋯ |  0.000000000 |  0.00000000 |  0.00000000 |  0.0000000 |  0.0000000 |  0.00000000 |  0.00000000 |  0.00000000 |  0.000000000 |  0.00000000 |\n",
       "| A2M |  0.040175335 |  0.05952886 |  0.025454407 |  0.07095019 |  0.1066928 |  0.03079474 |  0.1616818 |  0.079310732 |  0.1199986 |  0.03769264 | ⋯ |  0.052763272 |  0.08370718 |  0.06838003 |  0.1077101 |  0.0200616 |  0.04734121 |  0.09231839 |  0.04844618 |  0.005822723 |  0.07964579 |\n",
       "| A2M.AS1 |  7.224822540 |  7.02529435 |  5.042429225 | 12.28791629 |  5.7335597 |  7.13868477 |  9.2958055 |  7.779616698 | 10.7898108 |  6.35470746 | ⋯ |  4.052397341 |  2.16390527 |  4.32314182 |  9.3822274 |  4.4814680 |  6.95520729 |  2.50756773 |  9.90330650 |  3.730342348 |  8.72800822 |\n",
       "| A2ML1 |  0.000000000 |  0.00000000 |  0.004770096 |  0.00000000 |  0.0000000 |  0.01731259 |  0.0000000 |  0.005404598 |  0.0000000 |  0.00000000 | ⋯ |  0.008789078 |  0.00000000 |  0.00000000 |  0.0000000 |  0.0000000 |  0.00000000 |  0.00000000 |  0.00000000 |  0.004364659 |  0.00000000 |\n",
       "\n"
      ],
      "text/plain": [
       "         HTP0001B2    HTP0005A3   HTP0012A2    HTP0015A4   HTP0017A4 \n",
       "A1BG      1.213548628  0.74541369  0.599729304  0.32731643  1.3535745\n",
       "A1BG.AS1 38.222498659 24.92606386 26.427416614 12.07818145 24.1682646\n",
       "A1CF      0.003766156  0.00000000  0.003579259  0.00000000  0.0000000\n",
       "A2M       0.040175335  0.05952886  0.025454407  0.07095019  0.1066928\n",
       "A2M.AS1   7.224822540  7.02529435  5.042429225 12.28791629  5.7335597\n",
       "A2ML1     0.000000000  0.00000000  0.004770096  0.00000000  0.0000000\n",
       "         HTP0018B3   HTP0019B2  HTP0022B2    HTP0023A2  HTP0025A3   ⋯\n",
       "A1BG      0.94879940  0.6912469  1.097659107  0.6766134  0.27325337 ⋯\n",
       "A1BG.AS1 20.46200787 29.3621799 24.934104324 37.2498378 16.60207704 ⋯\n",
       "A1CF      0.00000000  0.0000000  0.000000000  0.0000000  0.00000000 ⋯\n",
       "A2M       0.03079474  0.1616818  0.079310732  0.1199986  0.03769264 ⋯\n",
       "A2M.AS1   7.13868477  9.2958055  7.779616698 10.7898108  6.35470746 ⋯\n",
       "A2ML1     0.01731259  0.0000000  0.005404598  0.0000000  0.00000000 ⋯\n",
       "         HTP0664A     HTP0665A    HTP0666A    HTP0667A   HTP0668A   HTP0669B   \n",
       "A1BG      0.255005402  0.52592489  0.28917116  0.4338034  1.0180578  1.07863098\n",
       "A1BG.AS1 32.137704564 37.67020602 19.36197036 23.5854179 29.6142852 31.45654523\n",
       "A1CF      0.000000000  0.00000000  0.00000000  0.0000000  0.0000000  0.00000000\n",
       "A2M       0.052763272  0.08370718  0.06838003  0.1077101  0.0200616  0.04734121\n",
       "A2M.AS1   4.052397341  2.16390527  4.32314182  9.3822274  4.4814680  6.95520729\n",
       "A2ML1     0.008789078  0.00000000  0.00000000  0.0000000  0.0000000  0.00000000\n",
       "         HTP0672A    HTP0676A    HTP0706A     HTP0708A   \n",
       "A1BG      0.66926351  1.31704224  0.422119153  1.10667053\n",
       "A1BG.AS1 17.50423671 26.70369761 22.246704141 31.47242852\n",
       "A1CF      0.00000000  0.00000000  0.000000000  0.00000000\n",
       "A2M       0.09231839  0.04844618  0.005822723  0.07964579\n",
       "A2M.AS1   2.50756773  9.90330650  3.730342348  8.72800822\n",
       "A2ML1     0.00000000  0.00000000  0.004364659  0.00000000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gtf_path <- here::here('data/GCF_000001405.38_GRCh38.p12_genomic.gtf')\n",
    "tpm_gene_counts_GSE190125=tpm_normalization(gene_counts_GSE190125, gtf_path)\n",
    "head(tpm_gene_counts_GSE190125)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1e94658-0ff9-412f-8531-072f7bd42e72",
   "metadata": {},
   "source": [
    "# PLIER preprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7c1e5485-6047-432c-be88-c9b7a4a08fa3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Computing SVD\n",
      "\n",
      "Done\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Assign arguments to variables \n",
    "output_file <- file.path(output_nb_path, 'htp_plier_prep.rds')\n",
    "                         \n",
    "# Load data\n",
    "expression_matrix <- tpm_gene_counts_GSE190125\n",
    "\n",
    "# Prepare output directory\n",
    "output_file_path=dirname(output_file)\n",
    "dir.create(dirname(output_file_path), showWarnings = FALSE, recursive = TRUE)\n",
    "\n",
    "# Combine the pathway data from PLIER\n",
    "all_paths <- PLIER::combinePaths(bloodCellMarkersIRISDMAP, svmMarkers, canonicalPathways, chemgenPathways, oncogenicPathways,\n",
    "                                 xCell, immunePathways, chr21_pathway)\n",
    "\n",
    "output_combine_allPaths_expressionMatrix = combine_allPaths_expressionMatrix(expression_matrix, all_paths)\n",
    "\n",
    "expression_matrix_cm=output_combine_allPaths_expressionMatrix$expression_matrix\n",
    "all_paths_cm=output_combine_allPaths_expressionMatrix$all_paths\n",
    "\n",
    "# compute rsvd/svd\n",
    "set.seed(123456)\n",
    "ns=ncol(expression_matrix_cm)\n",
    "message(\"Computing SVD\")\n",
    "set.seed(123456);svdres=rsvd(expression_matrix_cm, k=min(ns, max(200, ns/4)), q=3)\n",
    "message(\"Done\")\n",
    "\n",
    "# save z-scored expression data, the prior information matrix and svdres to be supplied to PLIER::PLIER and the number of PCs\n",
    "\n",
    "plier_data_list <- list(\"expression_matrix_cm\" = expression_matrix_cm,\n",
    "                        \"all_paths_cm\" = all_paths_cm,\n",
    "                        \"svdres\" = svdres)\n",
    "\n",
    "saveRDS(plier_data_list, file = output_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a2643ddb-7d56-433e-b414-ac47519d691b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>36362</li><li>400</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 36362\n",
       "\\item 400\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 36362\n",
       "2. 400\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 36362   400"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>36362</li><li>6623</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 36362\n",
       "\\item 6623\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 36362\n",
       "2. 6623\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 36362  6623"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(expression_matrix_cm)\n",
    "dim(all_paths_cm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "db040b15-3d9d-4c4b-8da2-66713af781d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 400 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>HTP0001B2</th><th scope=col>HTP0005A3</th><th scope=col>HTP0012A2</th><th scope=col>HTP0015A4</th><th scope=col>HTP0017A4</th><th scope=col>HTP0018B3</th><th scope=col>HTP0019B2</th><th scope=col>HTP0022B2</th><th scope=col>HTP0023A2</th><th scope=col>HTP0025A3</th><th scope=col>⋯</th><th scope=col>HTP0664A</th><th scope=col>HTP0665A</th><th scope=col>HTP0666A</th><th scope=col>HTP0667A</th><th scope=col>HTP0668A</th><th scope=col>HTP0669B</th><th scope=col>HTP0672A</th><th scope=col>HTP0676A</th><th scope=col>HTP0706A</th><th scope=col>HTP0708A</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GAS6</th><td>  1.6560099</td><td>  2.1402519</td><td>  1.261899</td><td>  1.7963980</td><td>  1.2777484</td><td>  2.1355887</td><td>  1.639557</td><td>  2.6667300</td><td>  1.0397606</td><td>  1.9595894</td><td>⋯</td><td>  2.2467221</td><td>  3.9415080</td><td>  1.5489576</td><td>  1.4465901</td><td>  3.2909590</td><td>  1.6876835</td><td>  1.9312305</td><td>  1.7675533</td><td>  3.0552640</td><td>  0.9686226</td></tr>\n",
       "\t<tr><th scope=row>MMP14</th><td>  2.4953244</td><td>  1.9930306</td><td>  1.199696</td><td>  2.5450903</td><td>  1.6079692</td><td>  2.3627318</td><td>  3.089905</td><td>  1.2012204</td><td>  2.4844051</td><td>  1.8453553</td><td>⋯</td><td>  1.8506383</td><td>  1.3945887</td><td>  1.8237727</td><td>  1.4166992</td><td>  1.5172404</td><td>  1.8680206</td><td>  2.0237556</td><td>  1.1416627</td><td>  3.3442387</td><td>  2.5316326</td></tr>\n",
       "\t<tr><th scope=row>MARCKSL1</th><td>276.7641962</td><td>270.4265478</td><td>279.941407</td><td>258.5501581</td><td>334.0244710</td><td>254.0435752</td><td>319.173642</td><td>395.2752429</td><td>331.1200619</td><td>288.8777938</td><td>⋯</td><td>391.0419482</td><td>259.6474733</td><td>490.4523672</td><td>430.3365181</td><td>266.8795789</td><td>326.3788623</td><td>309.2610658</td><td>286.2729580</td><td>432.2270856</td><td>351.7063226</td></tr>\n",
       "\t<tr><th scope=row>SPARC</th><td>  3.6546822</td><td> 57.8715760</td><td> 14.476117</td><td> 25.5837818</td><td> 26.2252392</td><td>  4.2739706</td><td> 47.820102</td><td> 37.2642962</td><td> 40.2125047</td><td> 32.8440224</td><td>⋯</td><td> 44.6884146</td><td> 11.1299801</td><td> 35.8952044</td><td> 66.0260550</td><td> 13.0404064</td><td> 25.0934466</td><td>  8.4555687</td><td>  6.9162273</td><td> 29.9525006</td><td> 23.0135713</td></tr>\n",
       "\t<tr><th scope=row>CTSD</th><td>112.4979271</td><td>173.5282446</td><td>191.496797</td><td>176.9456269</td><td>217.5920308</td><td>160.5192510</td><td>264.594733</td><td>201.8425316</td><td>194.9738538</td><td>172.2037147</td><td>⋯</td><td>224.8335890</td><td>240.7303094</td><td>428.5332534</td><td>407.3607449</td><td>206.2864302</td><td>222.3471960</td><td>115.9697302</td><td>161.9213684</td><td>303.9147606</td><td>149.5313723</td></tr>\n",
       "\t<tr><th scope=row>EPAS1</th><td>  0.5784673</td><td>  0.5292743</td><td>  0.733014</td><td>  0.4310616</td><td>  0.8043436</td><td>  0.5605247</td><td>  1.146536</td><td>  0.5680423</td><td>  0.2825346</td><td>  0.5529601</td><td>⋯</td><td>  0.3886176</td><td>  0.2061834</td><td>  0.2817493</td><td>  0.5104546</td><td>  0.3978614</td><td>  0.5328227</td><td>  0.6047124</td><td>  0.4014235</td><td>  0.6548923</td><td>  0.1767059</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 400 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & HTP0001B2 & HTP0005A3 & HTP0012A2 & HTP0015A4 & HTP0017A4 & HTP0018B3 & HTP0019B2 & HTP0022B2 & HTP0023A2 & HTP0025A3 & ⋯ & HTP0664A & HTP0665A & HTP0666A & HTP0667A & HTP0668A & HTP0669B & HTP0672A & HTP0676A & HTP0706A & HTP0708A\\\\\n",
       "\\hline\n",
       "\tGAS6 &   1.6560099 &   2.1402519 &   1.261899 &   1.7963980 &   1.2777484 &   2.1355887 &   1.639557 &   2.6667300 &   1.0397606 &   1.9595894 & ⋯ &   2.2467221 &   3.9415080 &   1.5489576 &   1.4465901 &   3.2909590 &   1.6876835 &   1.9312305 &   1.7675533 &   3.0552640 &   0.9686226\\\\\n",
       "\tMMP14 &   2.4953244 &   1.9930306 &   1.199696 &   2.5450903 &   1.6079692 &   2.3627318 &   3.089905 &   1.2012204 &   2.4844051 &   1.8453553 & ⋯ &   1.8506383 &   1.3945887 &   1.8237727 &   1.4166992 &   1.5172404 &   1.8680206 &   2.0237556 &   1.1416627 &   3.3442387 &   2.5316326\\\\\n",
       "\tMARCKSL1 & 276.7641962 & 270.4265478 & 279.941407 & 258.5501581 & 334.0244710 & 254.0435752 & 319.173642 & 395.2752429 & 331.1200619 & 288.8777938 & ⋯ & 391.0419482 & 259.6474733 & 490.4523672 & 430.3365181 & 266.8795789 & 326.3788623 & 309.2610658 & 286.2729580 & 432.2270856 & 351.7063226\\\\\n",
       "\tSPARC &   3.6546822 &  57.8715760 &  14.476117 &  25.5837818 &  26.2252392 &   4.2739706 &  47.820102 &  37.2642962 &  40.2125047 &  32.8440224 & ⋯ &  44.6884146 &  11.1299801 &  35.8952044 &  66.0260550 &  13.0404064 &  25.0934466 &   8.4555687 &   6.9162273 &  29.9525006 &  23.0135713\\\\\n",
       "\tCTSD & 112.4979271 & 173.5282446 & 191.496797 & 176.9456269 & 217.5920308 & 160.5192510 & 264.594733 & 201.8425316 & 194.9738538 & 172.2037147 & ⋯ & 224.8335890 & 240.7303094 & 428.5332534 & 407.3607449 & 206.2864302 & 222.3471960 & 115.9697302 & 161.9213684 & 303.9147606 & 149.5313723\\\\\n",
       "\tEPAS1 &   0.5784673 &   0.5292743 &   0.733014 &   0.4310616 &   0.8043436 &   0.5605247 &   1.146536 &   0.5680423 &   0.2825346 &   0.5529601 & ⋯ &   0.3886176 &   0.2061834 &   0.2817493 &   0.5104546 &   0.3978614 &   0.5328227 &   0.6047124 &   0.4014235 &   0.6548923 &   0.1767059\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 400 of type dbl\n",
       "\n",
       "| <!--/--> | HTP0001B2 | HTP0005A3 | HTP0012A2 | HTP0015A4 | HTP0017A4 | HTP0018B3 | HTP0019B2 | HTP0022B2 | HTP0023A2 | HTP0025A3 | ⋯ | HTP0664A | HTP0665A | HTP0666A | HTP0667A | HTP0668A | HTP0669B | HTP0672A | HTP0676A | HTP0706A | HTP0708A |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GAS6 |   1.6560099 |   2.1402519 |   1.261899 |   1.7963980 |   1.2777484 |   2.1355887 |   1.639557 |   2.6667300 |   1.0397606 |   1.9595894 | ⋯ |   2.2467221 |   3.9415080 |   1.5489576 |   1.4465901 |   3.2909590 |   1.6876835 |   1.9312305 |   1.7675533 |   3.0552640 |   0.9686226 |\n",
       "| MMP14 |   2.4953244 |   1.9930306 |   1.199696 |   2.5450903 |   1.6079692 |   2.3627318 |   3.089905 |   1.2012204 |   2.4844051 |   1.8453553 | ⋯ |   1.8506383 |   1.3945887 |   1.8237727 |   1.4166992 |   1.5172404 |   1.8680206 |   2.0237556 |   1.1416627 |   3.3442387 |   2.5316326 |\n",
       "| MARCKSL1 | 276.7641962 | 270.4265478 | 279.941407 | 258.5501581 | 334.0244710 | 254.0435752 | 319.173642 | 395.2752429 | 331.1200619 | 288.8777938 | ⋯ | 391.0419482 | 259.6474733 | 490.4523672 | 430.3365181 | 266.8795789 | 326.3788623 | 309.2610658 | 286.2729580 | 432.2270856 | 351.7063226 |\n",
       "| SPARC |   3.6546822 |  57.8715760 |  14.476117 |  25.5837818 |  26.2252392 |   4.2739706 |  47.820102 |  37.2642962 |  40.2125047 |  32.8440224 | ⋯ |  44.6884146 |  11.1299801 |  35.8952044 |  66.0260550 |  13.0404064 |  25.0934466 |   8.4555687 |   6.9162273 |  29.9525006 |  23.0135713 |\n",
       "| CTSD | 112.4979271 | 173.5282446 | 191.496797 | 176.9456269 | 217.5920308 | 160.5192510 | 264.594733 | 201.8425316 | 194.9738538 | 172.2037147 | ⋯ | 224.8335890 | 240.7303094 | 428.5332534 | 407.3607449 | 206.2864302 | 222.3471960 | 115.9697302 | 161.9213684 | 303.9147606 | 149.5313723 |\n",
       "| EPAS1 |   0.5784673 |   0.5292743 |   0.733014 |   0.4310616 |   0.8043436 |   0.5605247 |   1.146536 |   0.5680423 |   0.2825346 |   0.5529601 | ⋯ |   0.3886176 |   0.2061834 |   0.2817493 |   0.5104546 |   0.3978614 |   0.5328227 |   0.6047124 |   0.4014235 |   0.6548923 |   0.1767059 |\n",
       "\n"
      ],
      "text/plain": [
       "         HTP0001B2   HTP0005A3   HTP0012A2  HTP0015A4   HTP0017A4   HTP0018B3  \n",
       "GAS6       1.6560099   2.1402519   1.261899   1.7963980   1.2777484   2.1355887\n",
       "MMP14      2.4953244   1.9930306   1.199696   2.5450903   1.6079692   2.3627318\n",
       "MARCKSL1 276.7641962 270.4265478 279.941407 258.5501581 334.0244710 254.0435752\n",
       "SPARC      3.6546822  57.8715760  14.476117  25.5837818  26.2252392   4.2739706\n",
       "CTSD     112.4979271 173.5282446 191.496797 176.9456269 217.5920308 160.5192510\n",
       "EPAS1      0.5784673   0.5292743   0.733014   0.4310616   0.8043436   0.5605247\n",
       "         HTP0019B2  HTP0022B2   HTP0023A2   HTP0025A3   ⋯ HTP0664A   \n",
       "GAS6       1.639557   2.6667300   1.0397606   1.9595894 ⋯   2.2467221\n",
       "MMP14      3.089905   1.2012204   2.4844051   1.8453553 ⋯   1.8506383\n",
       "MARCKSL1 319.173642 395.2752429 331.1200619 288.8777938 ⋯ 391.0419482\n",
       "SPARC     47.820102  37.2642962  40.2125047  32.8440224 ⋯  44.6884146\n",
       "CTSD     264.594733 201.8425316 194.9738538 172.2037147 ⋯ 224.8335890\n",
       "EPAS1      1.146536   0.5680423   0.2825346   0.5529601 ⋯   0.3886176\n",
       "         HTP0665A    HTP0666A    HTP0667A    HTP0668A    HTP0669B   \n",
       "GAS6       3.9415080   1.5489576   1.4465901   3.2909590   1.6876835\n",
       "MMP14      1.3945887   1.8237727   1.4166992   1.5172404   1.8680206\n",
       "MARCKSL1 259.6474733 490.4523672 430.3365181 266.8795789 326.3788623\n",
       "SPARC     11.1299801  35.8952044  66.0260550  13.0404064  25.0934466\n",
       "CTSD     240.7303094 428.5332534 407.3607449 206.2864302 222.3471960\n",
       "EPAS1      0.2061834   0.2817493   0.5104546   0.3978614   0.5328227\n",
       "         HTP0672A    HTP0676A    HTP0706A    HTP0708A   \n",
       "GAS6       1.9312305   1.7675533   3.0552640   0.9686226\n",
       "MMP14      2.0237556   1.1416627   3.3442387   2.5316326\n",
       "MARCKSL1 309.2610658 286.2729580 432.2270856 351.7063226\n",
       "SPARC      8.4555687   6.9162273  29.9525006  23.0135713\n",
       "CTSD     115.9697302 161.9213684 303.9147606 149.5313723\n",
       "EPAS1      0.6047124   0.4014235   0.6548923   0.1767059"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_matrix_cm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0e67a6b3-f41d-4007-8366-a5b5eb0d73db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 6623 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>IRIS_Bcell-Memory_IgG_IgA</th><th scope=col>IRIS_Bcell-Memory_IgM</th><th scope=col>IRIS_Bcell-naive</th><th scope=col>IRIS_CD4Tcell-N0</th><th scope=col>IRIS_CD4Tcell-Th1-restimulated12hour</th><th scope=col>IRIS_CD4Tcell-Th1-restimulated48hour</th><th scope=col>IRIS_CD4Tcell-Th2-restimulated12hour</th><th scope=col>IRIS_CD4Tcell-Th2-restimulated48hour</th><th scope=col>IRIS_CD8Tcell-N0</th><th scope=col>IRIS_DendriticCell-Control</th><th scope=col>⋯</th><th scope=col>chr21_q21.1</th><th scope=col>chr21_q21.2</th><th scope=col>chr21_q21.3</th><th scope=col>chr21_q22.11</th><th scope=col>chr21_q22.12</th><th scope=col>chr21_q22.2</th><th scope=col>chr21_q22.13</th><th scope=col>chr21_q22.3</th><th scope=col>chr21_p12</th><th scope=col>chr21</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GAS6</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MMP14</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MARCKSL1</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SPARC</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>CTSD</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>EPAS1</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 6623 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & IRIS\\_Bcell-Memory\\_IgG\\_IgA & IRIS\\_Bcell-Memory\\_IgM & IRIS\\_Bcell-naive & IRIS\\_CD4Tcell-N0 & IRIS\\_CD4Tcell-Th1-restimulated12hour & IRIS\\_CD4Tcell-Th1-restimulated48hour & IRIS\\_CD4Tcell-Th2-restimulated12hour & IRIS\\_CD4Tcell-Th2-restimulated48hour & IRIS\\_CD8Tcell-N0 & IRIS\\_DendriticCell-Control & ⋯ & chr21\\_q21.1 & chr21\\_q21.2 & chr21\\_q21.3 & chr21\\_q22.11 & chr21\\_q22.12 & chr21\\_q22.2 & chr21\\_q22.13 & chr21\\_q22.3 & chr21\\_p12 & chr21\\\\\n",
       "\\hline\n",
       "\tGAS6 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tMMP14 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tMARCKSL1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tSPARC & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tCTSD & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tEPAS1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 6623 of type dbl\n",
       "\n",
       "| <!--/--> | IRIS_Bcell-Memory_IgG_IgA | IRIS_Bcell-Memory_IgM | IRIS_Bcell-naive | IRIS_CD4Tcell-N0 | IRIS_CD4Tcell-Th1-restimulated12hour | IRIS_CD4Tcell-Th1-restimulated48hour | IRIS_CD4Tcell-Th2-restimulated12hour | IRIS_CD4Tcell-Th2-restimulated48hour | IRIS_CD8Tcell-N0 | IRIS_DendriticCell-Control | ⋯ | chr21_q21.1 | chr21_q21.2 | chr21_q21.3 | chr21_q22.11 | chr21_q22.12 | chr21_q22.2 | chr21_q22.13 | chr21_q22.3 | chr21_p12 | chr21 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GAS6 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| MMP14 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| MARCKSL1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| SPARC | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| CTSD | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| EPAS1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "         IRIS_Bcell-Memory_IgG_IgA IRIS_Bcell-Memory_IgM IRIS_Bcell-naive\n",
       "GAS6     0                         0                     0               \n",
       "MMP14    0                         0                     0               \n",
       "MARCKSL1 0                         0                     0               \n",
       "SPARC    0                         0                     0               \n",
       "CTSD     0                         0                     0               \n",
       "EPAS1    0                         0                     0               \n",
       "         IRIS_CD4Tcell-N0 IRIS_CD4Tcell-Th1-restimulated12hour\n",
       "GAS6     0                0                                   \n",
       "MMP14    0                0                                   \n",
       "MARCKSL1 0                0                                   \n",
       "SPARC    0                0                                   \n",
       "CTSD     0                0                                   \n",
       "EPAS1    0                0                                   \n",
       "         IRIS_CD4Tcell-Th1-restimulated48hour\n",
       "GAS6     0                                   \n",
       "MMP14    0                                   \n",
       "MARCKSL1 0                                   \n",
       "SPARC    0                                   \n",
       "CTSD     0                                   \n",
       "EPAS1    0                                   \n",
       "         IRIS_CD4Tcell-Th2-restimulated12hour\n",
       "GAS6     0                                   \n",
       "MMP14    0                                   \n",
       "MARCKSL1 0                                   \n",
       "SPARC    0                                   \n",
       "CTSD     0                                   \n",
       "EPAS1    0                                   \n",
       "         IRIS_CD4Tcell-Th2-restimulated48hour IRIS_CD8Tcell-N0\n",
       "GAS6     0                                    0               \n",
       "MMP14    0                                    0               \n",
       "MARCKSL1 0                                    0               \n",
       "SPARC    0                                    0               \n",
       "CTSD     0                                    0               \n",
       "EPAS1    0                                    0               \n",
       "         IRIS_DendriticCell-Control ⋯ chr21_q21.1 chr21_q21.2 chr21_q21.3\n",
       "GAS6     1                          ⋯ 0           0           0          \n",
       "MMP14    0                          ⋯ 0           0           0          \n",
       "MARCKSL1 0                          ⋯ 0           0           0          \n",
       "SPARC    0                          ⋯ 0           0           0          \n",
       "CTSD     0                          ⋯ 0           0           0          \n",
       "EPAS1    1                          ⋯ 0           0           0          \n",
       "         chr21_q22.11 chr21_q22.12 chr21_q22.2 chr21_q22.13 chr21_q22.3\n",
       "GAS6     0            0            0           0            0          \n",
       "MMP14    0            0            0           0            0          \n",
       "MARCKSL1 0            0            0           0            0          \n",
       "SPARC    0            0            0           0            0          \n",
       "CTSD     0            0            0           0            0          \n",
       "EPAS1    0            0            0           0            0          \n",
       "         chr21_p12 chr21\n",
       "GAS6     0         0    \n",
       "MMP14    0         0    \n",
       "MARCKSL1 0         0    \n",
       "SPARC    0         0    \n",
       "CTSD     0         0    \n",
       "EPAS1    0         0    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(all_paths_cm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0846c03c-9932-4b0c-9cf0-01b0fc1eb1aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load data\n",
    "plier_data_list =  readRDS(file.path(output_nb_path, 'htp_plier_prep.rds'))\n",
    "expression_matrix_cm=plier_data_list$expression_matrix_cm\n",
    "all_paths_cm=plier_data_list$all_paths_cm\n",
    "svdres=plier_data_list$svdres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "205cbed3-664c-484f-be00-d27b4835a57a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "k is set to 195\n",
      "\n",
      "Removing 437 pathways with too few genes\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1810.413\n",
      "[1] \"L2 is set to 1810.41272823216\"\n",
      "[1] \"L1 is set to 905.206364116078\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "errorY (SVD based:best possible) = 10430\n",
      "\n",
      "New L3 is 0.00463091873353325\n",
      "\n",
      "New L3 is 0.00408677143846407\n",
      "\n",
      "New L3 is 0.00408677143846407\n",
      "\n",
      "New L3 is 0.00408677143846407\n",
      "\n",
      "New L3 is 0.00408677143846407\n",
      "\n",
      "New L3 is 0.00408677143846407\n",
      "\n",
      "New L3 is 0.00408677143846407\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#parameter_k <- c(0.25, 0.5, 0.75, 1, 1.25, 1.5, 1.75, 2)\n",
    "\n",
    "parameter_k <- c(1.25, 1.5, 1.75, 2)\n",
    "frac <- c(0.25, 0.5, 0.7, 1)\n",
    "\n",
    "for (param_k in parameter_k) {\n",
    "  for (f in frac) {\n",
    "    output_file <- file.path(output_nb_path, paste0('htp_plier_model_k', param_k, '_frac', f, '.rds'))\n",
    "\n",
    "    # Load data\n",
    "    expression_matrix_cm <- plier_data_list$expression_matrix_cm\n",
    "    all_paths_cm <- plier_data_list$all_paths_cm\n",
    "    svdres <- plier_data_list$svdres\n",
    "\n",
    "    # Compute k\n",
    "    k <- num.pc(svdres) * 2\n",
    "    k <- min(k, floor(ncol(expression_matrix_cm) * 0.9))\n",
    "    k <- k * param_k\n",
    "    k <- round(k, 0)\n",
    "    message(\"k is set to \", k)\n",
    "\n",
    "    # Run PLIER\n",
    "    plier_result <- PLIER::PLIER(data = expression_matrix_cm, priorMat = all_paths_cm, svdres = svdres, k = k, frac = f, scale = FALSE)\n",
    "\n",
    "    # Save results\n",
    "    saveRDS(plier_result, file = output_file)\n",
    "  }\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
